test/jdk/java/net/Socket/CloseOnFailureTest.java

```diff
@@ -0,0 +1,272 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import jdk.test.lib.Utils;
+import org.junit.function.ThrowingRunnable;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.function.ThrowingConsumer;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.VarHandle;
+import java.lang.reflect.Field;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.net.SocketAddress;
+import java.net.SocketImplFactory;
+import java.net.UnknownHostException;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertSame;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/*
+ * @test
+ * @bug 8343791
+ * @summary verifies the socket is closed on `connect()` failures
+ * @library /test/lib
+ * @run junit/othervm --add-opens java.base/java.net=ALL-UNNAMED CloseOnFailureTest
+ */
+class CloseOnFailureTest {
+
+    private static final VarHandle SOCKET_IMPL_FACTORY_HANDLE = createSocketImplFactoryHandle();
+
+    private static VarHandle createSocketImplFactoryHandle() {
+        try {
+            Field field = Socket.class.getDeclaredField("factory");
+            field.setAccessible(true);
+            return MethodHandles
+                    .privateLookupIn(Socket.class, MethodHandles.lookup())
+                    .findStaticVarHandle(Socket.class, "factory", SocketImplFactory.class);
+        } catch (NoSuchFieldException | IllegalAccessException error) {
+            throw new RuntimeException(error);
+        }
+    }
+
+    private static void withSocketImplFactory(SocketImplFactory newFactory, ThrowingRunnable runnable) throws Throwable {
+        SocketImplFactory oldFactory = (SocketImplFactory) SOCKET_IMPL_FACTORY_HANDLE.getAndSet(newFactory);
+        try {
+            runnable.run();
+        } finally {
+            SOCKET_IMPL_FACTORY_HANDLE.set(oldFactory);
+        }
+    }
+
+    private static final class MockSocketImpl extends ThrowingSocketImpl {
+
+        private final AtomicInteger closeInvocationCounter = new AtomicInteger(0);
+
```

Comment:
replacing Error with Exception in the vriable names allows for easier and quicker scanning of a line     

          private final Exception bindException;

         private final Exception connectException;

         private MockSocketImpl(Exception bindException, Exception connectException) {
             this.bindException = bindException;
             this.connectException = connectException;
         }

Reply:
Replaced all occurrences of `error` in variable names with `exception` in 35d680123f83b97b2a62fe99969aa0c18a75e6b1.